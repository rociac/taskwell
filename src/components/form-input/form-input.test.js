import FormInput from './form-input.component';
import renderer from 'react-test-renderer';
import { cleanup, render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

afterEach(cleanup);

describe('Form Input', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<FormInput />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders label if on props', () => {
    const { getByText } = render(<FormInput label="email" />);
    expect(getByText('email')).toBeInTheDocument();
  });

  it('should fire the onChange function', async () => {
    const handleChange = jest.fn();
    const { getByTestId } = render(<FormInput onChange={handleChange} />);
    await userEvent.type(getByTestId('input'), 'test');
    expect(handleChange).toHaveBeenCalledTimes(4);
  });
});