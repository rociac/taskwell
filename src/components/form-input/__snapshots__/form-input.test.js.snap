// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Input renders correctly 1`] = `
<div
  className="container"
>
  <input
    className="form-input"
    data-testid="input"
  />
</div>
`;
